# This patch removes the check for native_affinity for cross compiling.
# This patch has been pushed to htop on 23 November 2011, however, is
# not in the 1.0 release and may not be exactly what Hisham puts into
# the official build.
#
# Signed-off-by: Andy Kennedy <Andy.Kennedy@adtran.com>
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac     2011-11-20 20:46:48.000000000 -0600
+++ b/configure.ac     2011-11-23 10:41:44.000000000 -0600
@@ -111,7 +111,6 @@
 if test "$cross_compiling" = "no"; then
 AC_CHECK_FILE($PROCDIR/stat,,AC_MSG_ERROR(Cannot find /proc/stat. Make sure you have a Linux-compatible /proc filesystem mounted. See the file README for help.))
 AC_CHECK_FILE($PROCDIR/meminfo,,AC_MSG_ERROR(Cannot find /proc/meminfo. Make sure you have a Linux-compatible /proc filesystem mounted. See the file README for help.))
-fi
 
 AC_ARG_ENABLE(native_affinity, [AC_HELP_STRING([--enable-native-affinity], [enable native sched_setaffinity and sched_getaffinity for affinity support, disables hwloc])], ,enable_native_affinity="yes")
 if test "x$enable_native_affinity" = xyes; then
@@ -130,6 +129,7 @@
       AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])])
 fi
+fi
 
 AC_ARG_ENABLE(hwloc, [AC_HELP_STRING([--enable-hwloc], [enable hwloc support for CPU affinity])],, enable_hwloc="no")
 if test "x$enable_hwloc" = xyes
